client.go

```diff
@@ -1360,12 +1360,18 @@ func (cli *Client) uploadMediaToURL(data ReqUploadMedia) (*RespMediaUpload, erro
 		}
 	}
 
-	notifyURL := cli.BuildURL(MediaURLPath{"unstable", "fi.mau.msc2246", "upload", data.UnstableMXC.Homeserver, data.UnstableMXC.FileID, "complete"})
+	notifyURL, _ := url.Parse(cli.BuildURL(MediaURLPath{"unstable", "fi.mau.msc2246", "upload", data.UnstableMXC.Homeserver, data.UnstableMXC.FileID, "complete"}))
+
+	if len(data.FileName) > 0 {
+		q := notifyURL.Query()
+		q.Set("filename", data.FileName)
+		notifyURL.RawQuery = q.Encode()
+	}
```

Comment:
This should probably just use `BuildURLWithQuery` instead of re-parsing the URL manually

Reply:
https://github.com/mautrix/go/pull/94/commits/93c4f4a9725f9e80999b07435ce023a52de18ede